frontend/util.lua

```diff
@@ -301,6 +301,53 @@ function util.tableMerge(t1, t2)
     end
 end
 
+-- pairs(), but w/ keys sorted alphabetically
+-- c.f., http://lua-users.org/wiki/SortedIteration
+-- See also http://lua-users.org/wiki/SortedIterationSimple
+local function __genOrderedIndex( t )
+    local orderedIndex = {}
+    for key in pairs(t) do
+        table.insert( orderedIndex, key )
+    end
+    table.sort( orderedIndex )
+    return orderedIndex
+end
+
+local function orderedNext(t, state)
+    -- Equivalent of the next function, but returns the keys in the alphabetic
+    -- order. We use a temporary ordered key table that is stored in the
+    -- table being iterated.
+
+    local key = nil
+    --print("orderedNext: state = "..tostring(state) )
+    if state == nil then
+        -- the first time, generate the index
+        t.__orderedIndex = __genOrderedIndex( t )
+        key = t.__orderedIndex[1]
+    else
+        -- fetch the next value
+        for i = 1,table.getn(t.__orderedIndex) do
+            if t.__orderedIndex[i] == state then
+                key = t.__orderedIndex[i+1]
+            end
+        end
+    end
+
+    if key then
+        return key, t[key]
+    end
+
+    -- no more value to return, cleanup
+    t.__orderedIndex = nil
+    return
+end
+
+function util.orderedPairs(t)
```

Comment:
We have a function with the same name in base/ffi/util.lua :)
https://github.com/koreader/koreader-base/blob/b4124d64d4ea4403438f5a79e048f50e2f481f40/ffi/util.lua#L585-L632

Reply:
Oops :D.

I'll fix it later tonight ;).